MeshLib/MeshSearch/NodeSearch.cpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "NodeSearch.h"
+
+#include <set>
+
+#include <logog/include/logog.hpp>
+
+#include "MeshLib/Mesh.h"
+#include "MeshLib/Node.h"
+#include "MeshLib/Elements/Element.h"
+
+namespace MeshLib {
+
+NodeSearch::NodeSearch(const MeshLib::Mesh &mesh)
+	: _mesh(mesh)
+{
+}
+
+NodeSearch::~NodeSearch()
+{
+}
+
+std::size_t NodeSearch::searchByElementIDs(const std::vector<std::size_t> &elements, bool only_match_all_connected_elements)
```

Comment:
please review also this function. Second algorithm is added for the case `only_match_all_connected_elements==true`